### DISCUSSION TITLE
chore: Make examples versioned and more accessible

### DISCUSSION THREAD

**Discussion by asoorm** [TykTechnologies/tyk-operator#314] [Ideas]
It is common to see an `/examples` directory in the root of the project.
At the moment, our examples are hidden inside `/config/samples`.
This proposal allows promoting our vast set of examples so that they are more easily visible.

We must also consider versioning - such as when we introduce a new version of any custom resource.

```
/examples/v1alpha1
/examples/v1alpha2
```

We also need to consider how best to present examples which interact with other objects (which might also be versioned) including:

* Native objects such as Ingress
* The new Gateway API
* 3rd party services such as cert-manager

---

@gernest

**Comment by gernest** [comment]
This is a good idea. We should extend versioning to documentation as  well.
